dimes/dimes.py

```diff
@@ -2,25 +2,52 @@
 from pathlib import Path
 from typing import List, Union
 import warnings
+from dataclasses import dataclass
 
 from plotly.graph_objects import Figure, Scatter  # type: ignore
 from plotly.subplots import make_subplots  # type: ignore
 import koozie
 
 
+@dataclass
+class LineProperties:
+    color: Union[str, None] = None
+    line_type: Union[str, None] = None
+    marker_symbol: Union[str, None] = None
+    marker_size: Union[int, None] = None
+    marker_line_color: Union[str, None] = None
+    marker_fill_color: Union[str, None] = None
+    is_visible: bool = True
+
+
 class TimeSeriesData:
     """Time series data."""
 
+    class LineMode:
+        def determine_line_mode(
+            self, marker_symbol, marker_size, marker_line_color, marker_fill_color
+        ):
+            if all(
+                variables is None
+                for variables in (
+                    marker_size,
+                    marker_symbol,
+                    marker_line_color,
+                    marker_fill_color,
+                )
+            ):
+                return "lines"
+            else:
+                return "lines+markers"
+
```

Comment:
All of these variables are already data members of the LineProperties class. Why not add a method there instead?